tests/system/helper.py

```diff
@@ -0,0 +1,139 @@
+from datetime import datetime, timedelta
+import time
+from elasticsearch import Elasticsearch, NotFoundError
+from beat.beat import TimeoutError
+
+apm = "apm"
+apm_prefix = "{}*".format(apm)
+
+
+def wait_until(cond, max_timeout=10, poll_interval=0.25, name="cond"):
+    """
+    Like beat.beat.wait_until but catches exceptions
+    In a ElasticTest `cond` will usually be a query, and we need to keep retrying
+     eg. on 503 response codes
+    """
+    assert callable(cond), "First argument of wait_until must be a function"
+
+    start = datetime.now()
+    while datetime.now()-start < timedelta(seconds=max_timeout):
+        try:
+            result = cond()
+            if result:
+                return result
+        except AttributeError as ex:
+            raise ex
+        except Exception as ex:
+            pass
+        time.sleep(poll_interval)
+    raise TimeoutError("Timeout waiting for '{}' to be true. ".format(name) +
+                       "Waited {} seconds.".format(max_timeout))
+
+
+def cleanup(es, indices, templates, policies, indices_truncate, skip_pipelines=True):
```

Comment:
I'd suggest inverting `skip_pipelines` to `clean_pipelines`, so we don't have no stinkin' double negatives. Maybe it's worth making them all optional, and all verbs? e.g.

```suggestion
def cleanup(es, delete_indices=[], delete_templates=[], delete_policies=[], truncate_indices=[], delete_pipelines=[apm_prefix]):
```

And then use keyword args everywhere in callers, so it's clear what the args are.